message.go

```diff
@@ -2,9 +2,44 @@ package kafka
 
 import (
 	"bufio"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io/ioutil"
 	"time"
+
+	"github.com/eapache/go-xerial-snappy"
```

Comment:
Any reasons not to use [github.com/golang/snappy](https://godoc.org/github.com/golang/snappy) ?

Reply:
The Kafka encoding of snappy is using a specific Java library which does its own framing. If we really want to use `golang/snappy` I will have to dig into https://github.com/xerial/snappy-java